### PR TITLE
Add support for adding new Workspace Templates

### PR DESCRIPTION
**Pull request by sachinkundu**
# PR for issue #180 

1. If new current registration just add
2. If updating current for a previous registration mark previous current: false and save this one.
3. If not updating current just add this version if there is no conflict(same name and version)

Part 3/4 . Next up writing the repository tests and refactoring the if logic which is a bit clunky right now.

### CODE REVIEW

File: management_api_app/api/routes/workspace_templates.py [hunk 1]
```diff
@@ -35,15 +35,22 @@ async def create_workspace_template(workspace_template_create: WorkspaceTemplate
             template = workspace_template_repo.get_current_workspace_template_by_name(workspace_template_create.name)
         except EntityDoesNotExist:
             # first registration
-            pass
+            item_id = workspace_template_repo.create_workspace_template_item(workspace_template_create)
+            return WorkspaceTemplateIdInResponse(resourceTemplateId=item_id)
```
Comment (TessFerrandez) [inline management_api_app/api/routes/workspace_templates.py:39]: According to task - this should return the complete resource template created

File: management_api_app/db/repositories/workspace_templates.py [hunk 2]
```diff
@@ -40,3 +42,20 @@ def get_workspace_template_names(self) -> List[str]:
...
+
+    def create_workspace_template_item(self, workspace_template_create: WorkspaceTemplateInCreate):
+        item_id = str(uuid.uuid4())
+        resource_template = ResourceTemplate(
+            id=item_id,
+            name=workspace_template_create.name,
+            description=workspace_template_create.description,
+            version=workspace_template_create.version,
+            properties=workspace_template_create.parameters,
+            resourceType=workspace_template_create.resourceType,
```
Comment (TessFerrandez) [inline management_api_app/db/repositories/workspace_templates.py:54]: The resource type for workspace templates should always be workspace so this does not need to be in the input.

File: management_api_app/models/schemas/workspace_template.py [hunk 3]
```diff
@@ -42,8 +42,8 @@ class WorkspaceTemplateInCreate(BaseModel):
     name: str = Field(title="Name of workspace template")
     version: str = Field(title="Version of workspace template")
     description: str = Field(title=" Description of workspace template")
-    properties: dict = Field({}, title="Workspace template properties",
-                             description="Values for the properties required by the workspace template")
+    parameters: List[dict] = Field([{"location": "westeurope"}], title="Workspace template properties",
```
Comment (TessFerrandez) [inline management_api_app/models/schemas/workspace_template.py:45]: location is now azure_location

File: management_api_app/models/schemas/workspace_template.py [hunk 4]
```diff
@@ -53,7 +53,10 @@ class Config:
                 "name": "My workspace template",
                 "version": "0.0.1",
                 "description": "workspace template for great product",
-                "properties": {},
+                "properties": [{
+                    "name": "location",
```
Comment (TessFerrandez) [inline management_api_app/models/schemas/workspace_template.py:57]: same as last one: location is now changed to azure_location

**Review by TessFerrandez — APPROVED** [review]
approving with a few small change requests

**Review by TessFerrandez — APPROVED** [review]
approving with small change requests

File: management_api_app/models/schemas/workspace_template.py [hunk 5]
```diff
@@ -53,7 +53,10 @@ class Config:
                 "name": "My workspace template",
```
Comment (deniscep) [inline management_api_app/models/schemas/workspace_template.py:53]: Change example to be a docker image name (no spaces)
```suggestion
                "name": "my-tre-workspace",
```

File: management_api_app/db/repositories/workspace_templates.py (hunk 2 above)
Comment (sachinkundu) [reply management_api_app/db/repositories/workspace_templates.py:54]: fixing

File: management_api_app/models/schemas/workspace_template.py (hunk 3 above)
Comment (sachinkundu) [reply management_api_app/models/schemas/workspace_template.py:45]: changing

File: management_api_app/models/schemas/workspace_template.py (hunk 5 above)
Comment (sachinkundu) [reply management_api_app/models/schemas/workspace_template.py:53]: Good point

**Pull request merged**